lib/active_merchant/billing/gateways/flo2cash_rest.rb

```diff
@@ -0,0 +1,595 @@
+      #
+      # ==== Parameters
+      #
+      # * <tt>plan_id</tt> -- The identifier of Card Plan to change.
+      # * <tt>status</tt> -- can be "active | cancelled | suspended"
+      #   + Cancelled can only be applied to active/suspended plans
+      #   + Active can only be applied to suspended plans
+      #   + Suspended can only be applied to active plans
+      #
+      def update_card_plan(plan_id, status, options)
+        params = { plan_id: plan_id, status: status }
+        commit(:put, :update_card_plan, params, options)
+      end
+
+      # Update status of Direct Debit Plan +plan_id+ if possible
+      #
+      # ==== Parameters
+      #
+      # * <tt>plan_id</tt> -- The identifier of Direct Debit Plan to change.
+      # * <tt>status</tt> -- can be "cancelled | active | suspended"
+      #   + Cancelled can only be applied to active/suspended plans
+      #   + Active can only be applied to suspended plans
+      #   + Suspended can only be applied to active plans
+      #
+      def update_direct_debit_plan(plan_id, status, options)
+        params = { plan_id: plan_id, status: status }
+        commit(:put, :update_direct_debit_plan, params, options)
+      end
+
+      private
+
+      def commit(verb, action, params, options={})
+        begin
+          raw_response = ssl_request(verb, url(action, params), post_data(verb, params), headers(options))
+
+          response = parse(raw_response)
+          succeeded = success_from(response)
+        rescue ActiveMerchant::ResponseError => e
+          raise if e.response.code !~ CLIENT_ERROR_REGEX
+
+          response = parse(e.response.body)
+          succeeded = false
+        end
+
+        Response.new(
+          succeeded,
+          message_from(succeeded, response),
+          response,
+          authorization: authorization_from(action, response),
+          error_code: error_code_from(succeeded, response),
+          test: test?
+        )
+      end
+
+      def headers(options={})
+        {
+          'Content-Type' => 'application/json',
+          'Authorization' => "Basic #{options[:api_key]}"
+        }
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      end
+
+      def success_from(response)
+        !response.key?('status') || SUCCESS_STATUSES.include?(response['status'])
+      end
+
+      def message_from(succeeded, response)
+        succeeded ? 'Succeeded' : error_code_from(false, response)
+      end
+
+      # Returns a card token
+      #
+      # ==== Options
+      #
+      # * <tt>payment_method</tt> -- The instrument to charge.
+      # * <tt>options</tt> -- A hash of parameters.
+      #
+      def retrieve_token(payment_method, options)
+        if payment_method.is_a?(String)
+          payment_method
+        else
+          store(payment_method, options).authorization
+        end
+      end
+
+      def add_customer_resume(post, options)
+        post[:initiatedBy] = options[:initiatedBy]
```

Comment:
In order to keep the `options` keys more in a Ruby style, maybe make it `:initiated_by`. The `post` key is what the gateway expects, so that will have to be what it is but the options that we set, we can keep to a more ruby-esque kind of way.